### PR TITLE
Update http/https to v12

### PR DESCRIPTION
**Pull request by terurou**

### CODE REVIEW

File: src/js/node/http/Server.hx [hunk 1]
```diff
@@ -37,107 +37,168 @@ import js.Error;
...
 /**
-	HTTP server
+	This class inherits `from net.Server`.
 **/
 @:jsRequire("http", "Server")
 extern class Server extends js.node.net.Server {
 	/**
-		Limits maximum incoming headers count, equal to 1000 by default.
-		If set to 0 - no limit will be applied.
+			Limit the amount of time the parser will wait to receive the complete HTTP headers.
```
Comment (Gama11) [inline src/js/node/http/Server.hx]: Looks like there's too much indentation here?

I think running formatting it with haxe-formatter should take care of that - I started using it on hxnodejs in a recent commit.

**Comment by nadako** [comment]
Wow, thanks a lot!

File: src/js/node/http/Server.hx (hunk 1 above)
Comment (terurou) [reply src/js/node/http/Server.hx]: Thanks! I forgot to run the formatter.

Comment (Gama11) [reply src/js/node/http/Server.hx]: That reminds me, I should put formatOnSave into the VSCode config of this repo. :)

**Comment by Gama11** [comment]
Thanks for all of your contributions! hxnodejs would be in much worse shape without them. :)

Btw, do you have a good overview on where we are on https://github.com/HaxeFoundation/hxnodejs/issues/127 and https://github.com/HaxeFoundation/hxnodejs/issues/72?

**Comment by terurou** [comment]
> Btw, do you have a good overview on where we are on #127 and #72?
I'll write the "v12" overview in a few days.

**Pull request merged**